### PR TITLE
fix(viz): calendar heatmap granularity

### PR DESCRIPTION
**Pull request by maloun96**
### SUMMARY
Map subdomain to time_grain_sqla for creating query

## BEFORE
https://user-images.githubusercontent.com/8277264/108493764-5ffa1400-72af-11eb-92ee-22eec63dd8c5.mov

## AFTER 
https://user-images.githubusercontent.com/8277264/108493771-64263180-72af-11eb-9108-adb7a52e6a79.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/viz.py [hunk 1]
```diff
@@ -1031,7 +1030,7 @@ def get_data(self, df: pd.DataFrame) -> VizData:
         diff_secs = (end - start).total_seconds()
 
         if domain == "year":
-            range_ = diff_delta.years + 1
+            range_ = end.year - start.year + 1
```
Comment (maloun96) [inline superset/viz.py:1033]: Here for example I have 2 dates: 2020-08-01 and 2021-01-01 and the diff_delta.year will be "0" years and + 1 => range_ = 1 and will not display the data from "2021" year

File: superset/viz.py [hunk 2]
```diff
@@ -1053,6 +1052,15 @@ def query_obj(self) -> QueryObjectDict:
...
         d["metrics"] = fd.get("metrics")
+        mapping = {
+            "min": "PT1M",
+            "hour": "PT1H",
+            "day": "P1D",
+            "week": "P1W",
+            "month": "P1M",
+            "year": "P1Y",
+        }
+        d["extras"]["time_grain_sqla"] = mapping[fd.get("subdomain_granularity", "min")]
```
Comment (villebro) [inline superset/viz.py:1063]: Minor improvement proposal: To make this work for Legacy Druid, we'd need to set the `d["granularity"]` property to time grain. so something like this:
```python
time_grain = mapping[fd.get("subdomain_granularity", "min")]
if self.datasource.type == 'druid':
    d["granularity"] = time_grain
else:
    d["extras"]["time_grain_sqla"] = time_grain
```

File: superset/viz.py (hunk 1 above)
Comment (villebro) [reply superset/viz.py:1033]: good catch!

**Review by villebro — COMMENTED** [review]
Thanks for the fix! One minor proposal to make this work for both SQL and Druid, other than that LGTM

**Review by villebro — APPROVED** [review]

**Pull request merged**